utils/decode_kallsyms.py

```diff
@@ -0,0 +1,57 @@
+import bisect
+import re
+
+
+def binary_search_recursive(array, element):
+    return bisect.bisect_right(array, element, key=lambda y: int(y.split()[0], 16)) - 1
```

Comment:
Since you added the `kallsyms_search` variant, you can now remove the `binary_search_recursive` function.

Reply:
Aside from that, LGTM.

Reply:
Agree, need to remove the `binary_search_recursive` function as unused.

Reply:
yes, it was a mistake, i forgot to remove the function.. now it is removed